src/taskgraph/run-task/run-task

```diff
@@ -669,29 +685,18 @@ def git_checkout(
     # that consumers can compute the merge-base or files modified between the
     # two as needed.
     if base_rev and base_rev != NULL_REVISION:
-        args = ["git", "fetch", "origin", base_rev]
+        git_fetch(destination_path, base_rev, env=env)
 
-        retry_required_command(b"vcs", args, cwd=destination_path, extra_env=env)
-
-    # Fetch head_ref
-    args = ["git", "fetch"]
-    if head_ref and base_repo == head_repo:
-        # If a head_ref was provided, it might be tag, so we need to make sure we fetch
-        # those. This is explicitly only done when base and head repo match,
-        # because it is the only scenario where tags could be present. (PRs, for
-        # example, always include an explicit rev.) Failure to do this could result
-        # in not having a tag, or worse: having an outdated version of one.
-        # `--force` is needed to be able to update an existing tag.
-        args.extend(["--tags", "--force"])
-
-    else:
-        args.append("--no-tags")
+    # If a head_ref was provided, it might be tag, so we need to make sure we fetch
+    # those. This is explicitly only done when base and head repo match,
+    # because it is the only scenario where tags could be present. (PRs, for
+    # example, always include an explicit rev.) Failure to do this could result
+    # in not having a tag, or worse: having an outdated version of one.
+    tags = True if head_ref and base_repo == head_repo else False
```

Comment:
nit: `tags = head_ref and base_repo == head_repo` (although this gets modified in a following commit, so doesn't matter)